features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/lifecycle_api_router.jag

```diff
@@ -131,364 +131,405 @@ require('/modules/publisher.js').exec(function (ctx) {
+            try {
+                var permissionSet = registry.get(roleRestrictionPath);
+                var permissionContent = '' + permissionSet.content;
+                if (permissionContent) {
+                    userManager.denyRole('Internal/everyone', currentArtifact.path,
+                                         carbon.registry.actions.GET);
+                    userManager.denyRole('system/wso2.anonymous.role', currentArtifact.path,
+                                         carbon.registry.actions.GET);
+                }
+            } catch (e) {
+                log.debug('No role restrictions found for APP ' + uuid);
+                // continue work flow
+            }
+        }
+
+        // For all previously published web apps when publishing or re-publishing current web app,
+        // move subscriptions of that web app version to this web app version.
+        if ((appType == "webapp") && (previousArtifacts.length > 0)
+            && ((action == "Publish") || (action == "Re-Publish"))) {
+
+            for (i = 0; i < previousArtifacts.length; i++) {
+                operationResult = moveSubscriptions(previousArtifacts[i], currentArtifact, appType);
+                if (operationResult.isSuccess) {
+                    content.messages.push(operationResult.message);
+                } else {
+                    // some error occurred in moveSubscriptions function
+                    // abort further subscription moving
+                    break;
+                }
+            }
+            if (!operationResult.isSuccess) {
+                // something went wrong in the above loop, revert subscription moving
+                if (previousArtifacts.length == 1) {
+                    // if there are more than 1 previously published app,
+                    // then we cannot revert moved subscriptions :(
+                    moveSubscriptions(currentArtifact, previousArtifacts[0], appType);
+                    // ignore errors when reverting changes
+                }
+                // revert this app (current artifact) to unpublish state
+                promoteArtifactLifeCycle(artifactManager, currentArtifact, appType, "Unpublish");
+                // revert previously unpublished artifacts to Publish state
+                for (i = 0; i < previousArtifacts.length; i++) {
+                    promoteArtifactLifeCycle(artifactManager, previousArtifacts[i], appType,
+                                             "Re-Publish");
+                    // ignore errors when reverting changes
+                }
+                response.status = 500;
+                response.contentType = "application/json";
+                content.status = "Error";
+                content.messages.push(operationResult.message); // error message from last operation
+                response.content = stringify(content);
+                return;
+            }
+
+        }
+
+        // everything went great
```

Comment:
This comment was added to improve the readability. Since we don't have syntax-highlighting for Jaggery, I think its better have inline comments considering the length & complexity of this file.